### PR TITLE
Improve package documentation for goa and codegen packages

### PR DESCRIPTION
**Pull request by nitinmohan87**
Improves godoc documentation for the top-level `goa`, `eval`, and `codegen` packages.

### CODE REVIEW

File: codegen/funcs.go [hunk 1]
```diff
@@ -80,7 +63,7 @@ func Indent(s, prefix string) string {
 }
 
 // Add adds two integers and returns the sum of the two.
-func Add(a, b int) int { return a + b }
+//func Add(a, b int) int { return a + b }
```
Comment (raphael) [inline codegen/funcs.go]: Should that be removed entirely?

File: codegen/server/docs.go [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Package server contains code generation algorithms to produce an example
+// server and client implementation for the transports defined in the design.
```
Comment (raphael) [inline codegen/server/docs.go]: does package server really produce an example client implementation?

Comment (nitinmohan87) [reply codegen/server/docs.go]: I named the package after the `Server` expression. It produces both the example server and the client. Do you want to change the package name to something else?

Comment (raphael-rs) [reply codegen/server/docs.go]: We probably should yes. If the package generates the example code we could call it `example`?

**Pull request merged**